### DISCUSSION TITLE
Обработка несуществующих команд

### DISCUSSION THREAD

**Discussion by deadsik** [aiogram/aiogram#1604] [Q&A]
Hello,

Could you please let me know how to handle unrecognized commands?
There may be a solution, but I haven't found one.

For example, this would handle the command /start, 'My language English', or 'Моя мова Українська' (the user would press a button or enter it and receive a response):

```

@dp.message(CommandStart())
async def start(message: Message):
    await message.answer('Hello!\nPlease select a language', reply_markup=start_kb)

@dp.message(Command('My language English'))
async def help_command(message: Message):
        await message.answer('Welcome!', reply_markup=login_or_register_kb)

@dp.message(Command('Моя мова Українська'))
async def help_command(message: Message):
        await message.answer('Ласкаво просимо!', reply_markup=login_or_register_kb)
```

But if the user enters /test (for example), it will be ignored.

I thought of fixing it as follows:

```
async def start(message: Message):
    user_message = message.text
    if user_message == "/start":
        print("User enter start")
        await message.answer('Hello!\nPlease select a language', reply_markup=start_kb)

    elif user_message == 'My language English':
        print("User select English language")
        await message.answer('Welcome!', reply_markup=login_or_register_kb)

    elif user_message == 'Моя мова Українська':
        print("User select Russian language")
        await message.answer('Ласкаво просимо!', reply_markup=login_or_register_kb)

    else:
        print("user enter not found command")
        await message.answer('Command not found, run /start to restart the bot')
```

And it works this way, but I have a feeling that it's not the right approach, and there may be a way to implement it without such conditions (it should work as in the first example). Plus, if additional data needs to be received and processed from the user, this approach could lead to further complications.

**Comment by andrew000** [comment]
First of all you can't use `Command()` to handle simple text. Use `F.text` instead (Magic Filters).

To handle unrecognized commands - you can register handler with `Command()` filter and pass `re.Pattern` to it, to handle any command.
**❗️ This handler should be the last to be registered!**

***

FYP: You can use [aiogram-i18n](https://pypi.org/project/aiogram-i18n) to work with different locales and to handle text in different languages.

```python
@router.message(LF("my-language"))
async def handler(msg: Message, i18n: I18nContext) -> None:
    await msg.answer(i18n.your.language())
```
